pyproject.toml

```diff
@@ -215,8 +217,11 @@ max-doc-length = 120  # Match line-length
 # defaults when analyzing docstring sections.
 convention = "google"
 
-[tool.setuptools.dynamic.optional-dependencies.dev]
-file = ["dev-requirements.txt"]
+[tool.setuptools.dynamic.optional-dependencies]
+agents = {file = ["agent-requirements.txt"]}
+all = {file = ["agent-requirements.txt", "dev-requirements.txt", "google-requirements.txt"]}
```

Comment:
I don't really like `all` extras, because one always has to look up what that means. I would rather see our CI specifying `pip install -e .[agents,google,dev]`

Reply:
I vote to leave it for convenience, but we can specify in CI

Reply:
The thing is, it prevents us from just removing the requirements files entirely and directly writing dependencies in `pyproject.toml`. Also, all the requirements bloat the repo imo